pkgs/development/libraries/science/math/dealii/default.nix

```diff
@@ -0,0 +1,183 @@
+{ lib
+, stdenv
+, fetchzip
+, symlinkJoin
+, withDebug ? false
+, commonCxxFlags ? ""
+, debugCxxFlags ? ""
+, releaseCxxFlags ? ""
+, cmake
+, boost
+, withThreading ? true, tbb
+, withMPI ? true, mpi
+, with64bitIndices ? false
+, minimal ? false
+, doxygen
+, perl
+, graphviz
+, arpack
+, assimp
+, blas
+, lapack
+, gmsh
+, gsl
+, hdf5
+, hdf5-mpi
+, metis
+, muparser
+, p4est
+, p4est-sc
+, petsc
+, scalapack
+, symengine
+, zlib
+, suitesparse
+, openssh
+}:
+
+# Check PETSc compatibility
+assert withMPI -> petsc.mpiSupport;
+
+let
+  version = "9.4.0";
+
+  joinPkg = pkg: symlinkJoin {
+    name = "${pkg.name}-unsplit";
+    paths = pkg.all;
+  };
+  zlibJoined = joinPkg zlib;
+  boostJoined = joinPkg boost;
+  hdf5Joined = joinPkg (if withMPI then hdf5-mpi else hdf5);
+
+  # default PETSc in nixpkgs doesn't support 64-bit indices
+  petscEnabled = (!minimal) && (!with64bitIndices);
+
+  offlineDocs = fetchzip {
+    url = "https://dealii.43-1.org/downloads/dealii-${version}-offline_documentation.tar.gz";
+    sha256 = "sha256-368Dd8DCgoXw0vSyuDQYiDlZjNjVVf23RvyCX/RC9+4=";
+    stripRoot = false;
+  };
+
+  buildType = if withDebug then "DebugRelease" else "Release";
+in stdenv.mkDerivation rec {
+  pname = "dealii";
+  inherit version;
+
+  src = fetchzip {
+    url = "https://github.com/dealii/dealii/releases/download/v${version}/dealii-${version}.tar.gz";
```

Comment:
Why are we not fetching from git?

Reply:
I've read that `fetchzip` is now recommended instead of `fetchFrom*` for caching reasons. I'll change it then.